### PR TITLE
Adding new todo list

### PR DESCRIPTION
**Pull request by AlejandroHdzM**
Adding layout aplication and redirecting user if isn't logged in

### CODE REVIEW

File: .idea/.rakeTasks [hunk 1]
```diff
@@ -4,4 +4,4 @@ You are allowed to:
...
+--><RakeGroup description="" fullCmd="" taksId="rake" /></Settings>
```
Comment (diegosaul402) [inline .idea/.rakeTasks]: Any file from the `.idea` folder shouldn't be committed, please remove it from the PR.

**Comment by AlejandroHdzM** [comment]
## Adding TODO list option

## What implements/changes?
Adding layout, create new todo list with todo items on it and adding uer has_many todo_lists

## To manually test in Ruby console run the next commands:
u = User.last
### Setting a new todo list
u.todo_lists.create(tittle: "titulo", description: "description" date: "2018-12-12"

## Before - After screenshots
image

**Comment by pacocd** [comment]
@AlejandroHdzM just for the record, remember to add a title and a small description for your commits, in that way you can easily track when you added any change.

File: app/controllers/todo_items_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
```
Comment (pacocd) [inline app/controllers/todo_items_controller.rb]: is this comment required?

File: app/controllers/todo_items_controller.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+class TodoItemsController < ApplicationController
+  before_action :set_todo_list
+
+  def creat
```
Comment (pacocd) [inline app/controllers/todo_items_controller.rb]: did you mean create?

File: app/controllers/todo_lists_controller.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  # end
+
+  # GET /todo_lists/new
+  def new
+    @user = User.find(params[:user_id])
+    @todo_list = @user.todo_lists.new
+  end
+
+  # GET /todo_lists/1/edit
+  def edit; end
```
Comment (pacocd) [inline app/controllers/todo_lists_controller.rb]: any specific reason for using this syntax?

File: app/controllers/todo_lists_controller.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    @user = User.find(params[:user_id])
+    @todo_lists = @user.todo_lists
+  end
+
+  # GET /todo_lists/1
+  # GET /todo_lists/1.json
+  # def show
+  #   @user = User.find(params[:user_id])
+  #   @todo_list = @user.todo_lists.find(params[:id])
+  # end
```
Comment (pacocd) [inline app/controllers/todo_lists_controller.rb]: if this code is not required, it should be deleted

File: app/controllers/todo_lists_controller.rb [hunk 6]
```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+class TodoListsController < ApplicationController
+  before_action :authenticate_user!, :set_todo_list, only: [:show, :edit, :update, :destroy]
+
+  # GET /todo_lists
+  # GET /todo_lists.json
+  def index
+    @user = User.find(params[:user_id])
```
Comment (pacocd) [inline app/controllers/todo_lists_controller.rb]: I see that your repeating this line in some functions, you can refactor your code to follow this pattern and avoid repeating code:
``` ruby
class TodoListsController < ApplicationController
  attr_accesor :user
  before_action :define_user

  def new
    @todo_list = user.todo_lists.new
  end

  private def define_user
    user = User.find(params[:user_id])
  end

end```

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -10,6 +10,29 @@
...
+            <ul>
+              <li><%= link_to 'Profile', '#' %></li>
+              <li><%= link_to 'Settings', '#' %></li>
+              <li> <%= link_to 'Log out', destroy_user_session_path , method: :delete %> </li>
+            </ul>
+          </li>
+        <% else %>
+          <li><%= link_to 'Log in', user_session_path %></li>
+          <li><%= link_to 'Sign up', new_user_registration_path %></li>
+      <% end %>
```
Comment (pacocd) [inline app/views/layouts/application.html.erb]: @8geonirt @diegosaul402 is this a good practice? I mean, do devise validations in html code?

File: app/controllers/todo_items_controller.rb (hunk 2 above)
Comment (diegosaul402) [reply app/controllers/todo_items_controller.rb]: Yep @pacocd

File: app/controllers/todo_lists_controller.rb (hunk 4 above)
Comment (diegosaul402) [reply app/controllers/todo_lists_controller.rb]: Some people use it, but we don't encourage it

Comment (AlejandroHdzM) [reply app/controllers/todo_lists_controller.rb]: @pacocd Rubocop

File: app/controllers/todo_items_controller.rb (hunk 2 above)
Comment (pacocd) [reply app/controllers/todo_items_controller.rb]: oki. So, please dismiss this comment @AlejandroHdzM

File: app/controllers/todo_lists_controller.rb (hunk 4 above)
Comment (pacocd) [reply app/controllers/todo_lists_controller.rb]: so, if ```edit``` function is empty it should be removed

Comment (diegosaul402) [reply app/controllers/todo_lists_controller.rb]: Nope, is required but we prefer the two line style

**Pull request merged**